### PR TITLE
samples: wifi: shutdown: Clarify meaning of shutdown

### PR DESCRIPTION
**Pull request by krish2718**
Refer to the PS for definition of "shutdown".

### CODE REVIEW

File: samples/wifi/shutdown/README.rst [hunk 1]
```diff
@@ -28,6 +28,15 @@ The sample:
...
 #. Puts the nRF5340 SoC in the lowest possible power consumption mode.
 
+Shutdown mode
+*************
+
+The Wi-Fi chipset power state is automatically managed by the Wi-Fi driver, if the network interface is down the Wi-Fi chipset is put in `shutdown` mode.
+And if the network interface is up the Wi-Fi chipset is put in `active` mode.
+
+The `shutdown` here refers to de-asserting `buck_en` signal to the Wi-Fi chipset powering the Wi-Fi chipset down.
+See `the nRF70 Series power states`_ for more information.
```
Comment (ioannisg) [inline samples/wifi/shutdown/README.rst:38]: I think this needs to be moved up in the sample description @richabp  @krish2718 . We need to link the documentation when we first introduce the "shutdown" term. Note: in nRF70 Series HW terminology the Shutdown is a State, not a Mode.

My proposal at line 10:

"The Shutdown sample demonstrates how to put the Nordic Semiconductor's Wi-Fi® chipset in Shutdown state, where the device is completely powered OFF. See `the nRF70 Series power states`_ for more information on nRF70 Series power states."

File: samples/wifi/shutdown/README.rst [hunk 2]
```diff
@@ -28,6 +28,15 @@ The sample:
 #. Shuts down the Wi-Fi driver.
 #. Puts the nRF5340 SoC in the lowest possible power consumption mode.
 
+Shutdown mode
```
Comment (ioannisg) [inline samples/wifi/shutdown/README.rst:31]: Let's remove this title, as it is confusing with the related power state. If you want a separate section just use

Notes
******
 or any other generic name :)

File: samples/wifi/shutdown/README.rst [hunk 3]
```diff
@@ -28,6 +28,15 @@ The sample:
...
+*************
+
+The Wi-Fi chipset power state is automatically managed by the Wi-Fi driver, if the network interface is down the Wi-Fi chipset is put in `shutdown` mode.
```
Comment (ioannisg) [inline samples/wifi/shutdown/README.rst:34]: This is exactly what we need to have here :) I suggest breaking down in smaller parts and being more verbose. 
My proposal, based on offline info from @krish2718 :

"The nRF70 Series transition to and from Shutdown state is automatically managed by the Wi-Fi driver. When the network interface (net_if: or @krish2718  , you can use an "official" term here which describes the zephyr network interface), is brought down, the Wi-Fi driver puts the nRF70 Series device in `Shutdown` power state. When the network interface is brought up, the Wi-Fi driver puts the nRF70 Series device in `Active` state.

File: samples/wifi/shutdown/README.rst [hunk 4]
```diff
@@ -28,6 +28,15 @@ The sample:
...
+And if the network interface is up the Wi-Fi chipset is put in `active` mode.
+
+The `shutdown` here refers to de-asserting `buck_en` signal to the Wi-Fi chipset powering the Wi-Fi chipset down.
```
Comment (ioannisg) [inline samples/wifi/shutdown/README.rst:37]: Let's make this a bit more verbose:
"As part of transition to `Shutdown` state, the Wi-Fi driver de-asserts the `buck_en` signal to the nRF70 Series chipset to completely power down the device. When requested to power on the device, the Wi-Fi driver will re-assert the `buck_en` signal to the nRF70 Series chipset."

Comment (ioannisg) [reply samples/wifi/shutdown/README.rst:37]: And I would also add a note, of the impact of power down, when it comes to the ram retention.

Proposal for a "note" @krish2718 @richabp 
"In Shutdown state the nRF70 Series device does not retain any state information. Powering up the device after Shutdown invokes the complete nRF70 Series boot up sequence."

**Review by ioannisg — CHANGES_REQUESTED** [review]
@krish2718 I made some significant changes. Please see if my proposals make sense and whether they are aligned with the software behavior :)

File: samples/wifi/shutdown/README.rst (hunk 4 above)
Comment (krish2718) [reply samples/wifi/shutdown/README.rst:37]: Isn't this covered in PS?

Comment (ioannisg) [reply samples/wifi/shutdown/README.rst:37]: It is actually covered even in the NCS docs (Operating on power save). But it does not harm adding a note here. MY point is to make clear that as part of the transition from Shutdown to Active (via net if up, or whatever...) the Wi-Fi driver will perform the proper boot up. This is not covered in PS, as this is mostly SW-related info. We may have it on the "Working with nRF70 Series", but a repetition in this sample does not harm.

Comment (krish2718) [reply samples/wifi/shutdown/README.rst:37]: I don't want to repeat details instead re-use, so, buck-en stuff is covered in PS and I doubt users will understand that without additional context which again is available in the PS.

File: samples/wifi/shutdown/sample.yaml [hunk 5]
```diff
@@ -17,3 +17,11 @@ tests:
       - nrf52840dk_nrf52840
     platform_allow: nrf5340dk_nrf5340_cpuapp nrf52840dk_nrf52840
     tags: ci_build
+  sample.nrf7000_location.shutdown:
+    build_only: true
+    extra_args: SHIELD=nrf7002ek
```
Comment (ioannisg) [inline samples/wifi/shutdown/sample.yaml:22]: should the shield here be the emulated nRF7000 target @krish2718 ?

File: samples/wifi/shutdown/README.rst [hunk 6]
```diff
@@ -7,7 +7,9 @@ Wi-Fi: Shutdown
    :local:
    :depth: 2
 
-The Shutdown sample demonstrates how to put the Nordic Semiconductor's Wi-Fi® chipset in shutdown mode.
+The Shutdown sample demonstrates how to put the Nordic Semiconductor's Wi-Fi® chipset in shutdown state, where the device is completely powered OFF.
```
Comment (ioannisg) [inline samples/wifi/shutdown/README.rst:10]: Please, capitalize "Shutdown state" here.

File: samples/wifi/shutdown/README.rst (hunk 4 above)
Comment (ioannisg) [reply samples/wifi/shutdown/README.rst]: Based on your last comment: how about removing also the above sentence and putting this in the Power States section, in the "Operating in Power Save Modes" chapter in "Working with nRF70 Series"?

File: samples/wifi/shutdown/README.rst [hunk 7]
```diff
@@ -28,6 +33,7 @@ The sample:
 #. Shuts down the Wi-Fi driver.
```
Comment (ioannisg) [inline samples/wifi/shutdown/README.rst:33]: Can we be more verbose here? Proposal:
"Brings down the Wi-Fi network interface, which automatically directs the Wi-Fi driver to power down the nRF70 device.

File: samples/wifi/shutdown/README.rst [hunk 8]
```diff
@@ -20,6 +22,9 @@ The sample supports the following development kits:
 Overview
 ********
 
+The nRF70 Series transition to and from Shutdown state is automatically managed by the Wi-Fi driver. When the network interface is brought down, the Wi-Fi driver puts
```
[the remark this answers is not in the document]
Comment (krish2718) [reply samples/wifi/shutdown/README.rst:25]: But isn't this the main point that we are trying to clarify here?

File: samples/wifi/shutdown/README.rst (hunk 4 above)
Comment (krish2718) [reply samples/wifi/shutdown/README.rst]: That was the main intention of the PR as the source code doesn't have any `poweroff` API, so, just a line to tie power with iface.

**Review by ioannisg — APPROVED** [review]

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 9]
```diff
@@ -29,6 +29,11 @@ The nRF70 Series device can be in one of the following power states:
   In this state, the device consumes very low power (~2 µA) and does not retain any state information (apart from the values in the OTP memory).
   The device will only respond to a BUCKEN assertion to wake from the Shutdown state.
 
+The nRF70 Series transition to and from Shutdown state is automatically managed by the Wi-Fi driver. When the network interface is brought down, the Wi-Fi driver puts
+the nRF70 Series device in Shutdown power state. When the network interface is brought up, the Wi-Fi driver puts the nRF70 Series device in Active state.
```
Comment (richabp) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:33]:
```suggestion
The nRF70 Series transition to and from the Shutdown state is automatically managed by the Wi-Fi driver.
When the network interface is brought down, the driver puts the nRF70 Series device in Shutdown power state.
When the network interface is brought up, the driver puts the nRF70 Series device in Active state.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 10]
```diff
@@ -29,6 +29,11 @@ The nRF70 Series device can be in one of the following power states:
...
+
+.. note:: The application needs to re-initialize the Wi-Fi association after the network interface is brought up.
```
Comment (richabp) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:35]:
```suggestion
.. note::
   The application needs to reinitialize the Wi-Fi association after the network interface is brought up.
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Comment by ioannisg** [comment]
Interestingly, the whole documentation of this sample is tailored around the nRF5340 host, although the sample is supported on nRF52840 and nRF91 Series. Assuming that the selection of the host is just an example (that is we are confident that this sample can also put the nrf52840 or the nrf9160 in lowest power mode) could we reflect that in the example functionality description?

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 11]
```diff
@@ -29,6 +29,13 @@ The nRF70 Series device can be in one of the following power states:
   In this state, the device consumes very low power (~2 µA) and does not retain any state information (apart from the values in the OTP memory).
   The device will only respond to a BUCKEN assertion to wake from the Shutdown state.
 
+The nRF70 Series transition to and from Shutdown state is automatically managed by the Wi-Fi driver.
```
Comment (richabp) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:32]:
```suggestion
The nRF70 Series transition to and from the Shutdown state is automatically managed by the Wi-Fi driver.
```

File: doc/nrf/links.txt [hunk 12]
```diff
@@ -566,6 +566,8 @@
 
 .. _`Guidelines and application notes for nRF70 Series devices`: https://infocenter.nordicsemi.com/topic/struct_nrf70/struct/nrf70_guidelines.html
 
+.. _`the nRF70 Series power states`: https://infocenter.nordicsemi.com/index.jsp?topic=%2Fps_nrf7002%2Fchapters%2Ffunctional%2Fdoc%2Fpower_states.html&cp=3_0_0_4_0
```
[the remark this answers is not in the document]
Comment (richabp) [reply doc/nrf/links.txt:569]: This seems to be missed.

File: samples/wifi/shutdown/README.rst [hunk 13]
```diff
@@ -7,8 +7,10 @@ Wi-Fi: Shutdown
    :local:
    :depth: 2
 
-The Shutdown sample demonstrates how to put the Nordic Semiconductor's Wi-Fi® chipset in shutdown mode.
-This also demonstrates how to achieve the lowest possible power consumption in the nRF5340 SoC when Wi-Fi is enabled but not in use.
+The Shutdown sample demonstrates how to put the Nordic Semiconductor's Wi-Fi® chipset in the Shutdown state, where the device is completely powered off.
+For more information, see the `the nRF70 Series power states`_ page.
```
Comment (richabp) [inline samples/wifi/shutdown/README.rst:11]:
```suggestion
For more information, see the `nRF70 Series power states`_ page.
```

File: samples/wifi/shutdown/README.rst [hunk 14]
```diff
@@ -7,8 +7,10 @@ Wi-Fi: Shutdown
...
+
+This also demonstrates how to achieve the lowest possible power consumption in the Host SoC (nRF53, nRF52, nRF91 etc.) when Wi-Fi is enabled but not in use.
```
Comment (richabp) [inline samples/wifi/shutdown/README.rst:13]:
```suggestion
This also demonstrates how to achieve the lowest possible power consumption in the Host SoC (nRF52, nRF53, or nRF91 Series) when Wi-Fi is enabled but not being used.
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by richabp — APPROVED** [review]

**Pull request merged**